llvm/lib/Transforms/IPO/FunctionAttrs.cpp

```diff
@@ -633,10 +633,13 @@ ArgumentAccessInfo getArgmentAccessInfo(const Instruction *I,
       [](Value *Length,
          std::optional<int64_t> Offset) -> std::optional<ConstantRange> {
     auto *ConstantLength = dyn_cast<ConstantInt>(Length);
-    if (ConstantLength && Offset)
+    if (ConstantLength && Offset) {
```

Comment:
Include in the if condition?
```suggestion
    if (ConstantLength && Offset && ConstantLength->isNonNegative()) {
```

Reply:
Done, thanks!